index.bs

```diff
@@ -7652,6 +7652,122 @@ To <dfn abstract-op>Create a new supplemental public key record</dfn>, perform t
         [=set/append=] this [=supplemental public key record=] to |credentialRecord|.[$credential record/supplementalPubKeys$].
 
 
+### Confirmation Extension (confirmation) ### {#sctn-confirmation-extension}
+
+This extension allows for capturing user confirmation. A
+   [=[RP]=] can specify a confirmation prompt string, intended for display by the platform and by the authenticator (if supported).
+   With this approach, [=Relying Parties=] can use the feature independently of the capabilities of the authenticator used by the user,
+   while still benefitting from the increased security level if the authenticator itself supports showing the confirmation prompt.
+
+Authenticators could use a Trusted UI to ensure that the user indeed sees the confirmation prompt.
+
+Example uses cases could be "I want to move $1234 from account A to account B" or "I want to share my health data with hospital X".
+
+: Extension identifier
+:: `confirmation`
+
+: Operation applicability
+:: [=authentication extension|Authentication=]
+
+: Client extension input
+:: A single USVString confirmationPrompt.  This string might contain the following formatting tags: <code>&lt;b&gt;&lt;/b&gt;</code> to mark the text inbetween as bold and <code>&lt;br /&gt;</code> to force a line break.  Support of these formatting tags is "best effort".  Implementations not supporting it SHALL NOT display the tags in "verbatim".
+
+    <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+      USVString confirmationPrompt;
+    };
+    </xmp>
+
+: Client extension processing
+:: If this extension is present, the client SHALL
+       1. use a dialog to the user that makes the user aware of the confirmation to be provided (as opposed to doing a simple sign in).
+       1. display the confirmation prompt to the user. The client SHOULD
+             indicate that the confirmation prompt originates from a specific relying party
+             (as opposed to the platform itself).
+     
+       1. use the {{CollectedClientConfirmationData}} structure containing the confirmation prompt instead of using
+             the {{CollectedClientData}} structure.
+     
+       1. pass-through the extension to the authenticator (see "client extension output" below)
+       1. pass-through the "authenticator extension output" to the caller as part of the assertion
```

Comment:
"If this extension is present, the client SHALL" is implicit and can be dropped. The last step is also implicit in the "Client extension output" section. Please begin sentences with an uppercase letter and use the `[=[RP]=]` term.

I would also write the fourth step as "set the authenticator extension input to...":

```suggestion
::  1. Use a dialog to the user that makes the user aware of the confirmation to be provided (as opposed to doing a simple sign in).
    1. Display the confirmation prompt to the user. The client SHOULD
        indicate that the confirmation prompt originates from a specific [=[RP]=]
        (as opposed to the platform itself).
     
    1. Use the {{CollectedClientConfirmationData}} structure containing the confirmation prompt instead of using
        the {{CollectedClientData}} structure.
    1. Set the `confirmation` [=authenticator extension input=] to the value of {{AuthenticationExtensionsClientInputs/confirmation}}, encoded as a CBOR text string.
```

Reply:
done